mmcv/runner/hooks/profiler.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Open-MMLab. All rights reserved.
+import warnings
+from typing import Callable, List, Optional, Union
+
+import torch
+
+from ..dist_utils import master_only
+from .hook import HOOKS, Hook
+
+
+@HOOKS.register_module()
+class ProfilerHook(Hook):
+    """Profiler context manager.
```

Comment:
`"context manager"` may not be accurate. Maybe profiler to analyze training process.